### PR TITLE
Enable Perf JITEvents by default

### PR DESCRIPTION
**Pull request by vchuravy**
Yggdrasil is already shipping with this integration, and so it would be nice to have this available on release builds.
I also added the option to enable GDB's JIT integration on normal builds.

For the perf support to work well one probably needs a very recent `perf`.

https://github.com/dotnet/coreclr/pull/26897#issuecomment-691195475

### CODE REVIEW

**Comment by vchuravy** [comment]
any objections to this?

**Comment by maleadt** [comment]
Any reason not to just enable the GDB listener as soon as you use `-g2` or higher? An environment variable seems clumsy, and with `-g2` you're explicitly asking for for the JIT to generate more debug info (whereas a debug builds reflects more on the debuggability of the `julia` binary).

**Comment by vchuravy** [comment]
I think the issue is that these listeners will clutter your harddrive. Generally I think turning it on for `-g 2` would be a good idea.

**Review by vtjnash — APPROVED** [review]

File: Make.inc [hunk 1]
```diff
@@ -82,7 +82,7 @@ endif
 USE_OPROFILE_JITEVENTS ?= 0
 
 # Set to 1 to enable profiling with perf
-USE_PERF_JITEVENTS ?= 0
+USE_PERF_JITEVENTS ?= 1
```
Comment (vtjnash) [inline Make.inc:85]: We should have made this conditional on `ifeq ($(OS),Linux)`, since LLVM now refuses to build on non-linux

Comment (vchuravy) [reply Make.inc:85]: Mea culpa, https://github.com/JuliaPackaging/Yggdrasil/blob/80d832c7c02bcb52e787a883ee312e1d8eae223a/L/LLVM/common.jl#L150

Will push a fix.

**Pull request merged**